sdk/src/azure/platform/az_posix.c

```diff
@@ -26,3 +30,129 @@ AZ_NODISCARD az_result az_platform_sleep_msec(int32_t milliseconds)
   (void)usleep((useconds_t)milliseconds * _az_TIME_MICROSECONDS_PER_MILLISECOND);
   return AZ_OK;
 }
+
+void az_platform_critical_error()
+{
+  while (1)
+  {
+  }
+}
+
+AZ_NODISCARD az_result az_platform_get_random(int32_t* out_random)
+{
+  _az_PRECONDITION_NOT_NULL(out_random);
+  
+  *out_random = (int32_t)random();
+  return AZ_OK;
+}
+
+void _thread_handler(union sigval sv)
+{
+  az_platform_timer* timer_handle = sv.sival_ptr;
+
+  _az_PRECONDITION_NOT_NULL(timer_handle);
+  _az_PRECONDITION_NOT_NULL(timer_handle->_internal.callback);
+  timer_handle->_internal.callback(timer_handle->_internal.sdk_data);
+}
+
+AZ_NODISCARD az_result az_platform_timer_create(
+    az_platform_timer* timer_handle,
+    az_platform_timer_callback callback,
```

Comment:
validate callback is not null?

Reply:
+1 here and throughout the impl - please make sure all platform APIs have preconditions.